src/main/java/data_access/PodcastDataAccessObject.java

```diff
@@ -0,0 +1,124 @@
+package data_access;
+
+import entities.*;
+import java.io.*;
+import java.net.URISyntaxException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class PodcastDataAccessObject implements PodcastDataAccess {
+
+    private final Map<UUID, Podcast> podcastMap;
+    private final EpisodeDataAccess episodeDAO;
+
+    public PodcastDataAccessObject(EpisodeDataAccess episodeDAO) {
+        podcastMap = new HashMap<>();
+        this.episodeDAO = episodeDAO;
+        loadPodcasts();
+    }
+
+    @Override
+    public List<MediaItem> getEpisodesForPodcast(UUID podcastId) {
+        Podcast podcast = podcastMap.get(podcastId);
+        return podcast.getItems();
+    }
+
+    @Override
+    public Podcast getPodcastById(UUID id) {
+        return podcastMap.get(id);
+    }
+
+    @Override
+    public boolean savePodcast(Podcast podcast) {
+        podcastMap.put(podcast.getId(), podcast);
+        return save();
+    }
+
+    private void loadPodcasts() {
```

Comment:
You're going to have a bug if the podcast name has a comma in it. I fixed this for the episode DAO.